operator/pkg/controllers/etcd/etcd.go

```diff
@@ -20,181 +20,77 @@ import (
 	"fmt"
 	"net"
 
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/awslabs/kit/operator/pkg/apis/infrastructure/v1alpha1"
-	pkiutil "github.com/awslabs/kit/operator/pkg/pki"
-	"github.com/awslabs/kit/operator/pkg/utils/patch"
+	"github.com/awslabs/kit/operator/pkg/kubeprovider"
+	"github.com/awslabs/kit/operator/pkg/utils/common"
+	"github.com/awslabs/kit/operator/pkg/utils/object"
 	"github.com/awslabs/kit/operator/pkg/utils/secrets"
 
 	"go.uber.org/zap"
-	appsv1 "k8s.io/api/apps/v1"
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/types"
 	certutil "k8s.io/client-go/util/cert"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
 const (
 	etcdRootCACommonName = "etcd/ca"
 )
 
 type Controller struct {
-	kubeClient      client.Client
-	secretsProvider *secrets.Provider
+	kubeClient   *kubeprovider.Client
+	certificates *common.CertificatesProvider
 }
 
-func New(kubeclient client.Client) *Controller {
-	return &Controller{kubeClient: kubeclient, secretsProvider: secrets.New(kubeclient)}
-}
+type reconciler func(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (err error)
 
-func (c *Controller) Reconcile(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (err error) {
-	// generate etcd service object
-	if err := c.createService(ctx, controlPlane); err != nil {
-		return err
-	}
-	// Create ETCD certs and keys, store them as secret in the management server
-	if err := c.createSecrets(ctx, controlPlane); err != nil {
-		return err
-	}
-	// create etcd stateful set
-	if err := c.createStatefulset(ctx, controlPlane); err != nil {
-		return err
-	}
-	return nil
+func New(kubeclient *kubeprovider.Client) *Controller {
+	return &Controller{kubeClient: kubeclient, certificates: common.New(kubeclient)}
```

Comment:
common.New feels a bit off to me. Aren't all of our controllers using the same kubeprovider.Client object?